src/werkzeug/local.py

```diff
@@ -336,7 +336,6 @@ def __delitem__(self, key):
     __lt__ = lambda x, o: x._get_current_object() < o
     __le__ = lambda x, o: x._get_current_object() <= o
     __eq__ = lambda x, o: x._get_current_object() == o
-    __ne__ = lambda x, o: x._get_current_object() != o
```

Comment:
Can't remove this, it needs to proxy to whatever the target defines.

Reply:
Thanks, added this back.

I'd incorrectly thought that we can safely assume the proxied object defines `__eq__` such that `not proxied_obj.__eq__(other) == proxied_obj.__ne__(other)`.

(To try to prove it, I added a couple tests here to exercise this, as I noticed `__ne__` was not getting called from the tests before, and verified that I could remove this `__ne__` without causing a failure.)

But now I agree it's incorrect not to proxy the `__ne__` call through; we should remain faithful to whatever the proxied object's `__ne__` does.